x-pack/legacy/plugins/apm/public/components/app/TransactionDetails/WaterfallWithSummmary/WaterfallContainer/Waterfall/waterfall_helpers/waterfall_helpers.ts

```diff
@@ -69,17 +47,16 @@ interface IWaterfallItemBase {
    * skew from timestamp in us
    */
   skew: number;
-  childIds?: Array<IWaterfallItemBase['id']>;
 }
 
 interface IWaterfallItemTransaction extends IWaterfallItemBase {
-  transaction: Transaction;
+  trace: Transaction; // TODO: check this name
```

Comment:
@sqren I removed most of the properties from `getTransactionItem ` and `getSpanItem`, and replaced it by the real Transaction and Span. But I changed the name to `trace` because there are some common properties between them. So I don't have to check the type every time. I can simply do `item.trace.timestamp.us`.
I'd like to have your insights about this change and about the name `trace`, don't know if this is the best.

Reply:
I think `doc` is better than `trace`. `trace` is normally a list (tree?) or transactions/spans. Whereas this is the actual elasticsearch document for either a span or a transaction.